spec/models/friendship_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe Friendship, type: :model do
+  subject { build(:friendship) }
+  let(:user) { create(:user) }
+  let(:friend) { create(:user) }
+  let(:friends) { create(:friendship) }
+
+  it 'should be valid' do
```

Comment:
I think our friendship_spec is a bit unorganized. Here the things we normally wanna test for models:

1. Validations - test for valid instance and for an invalid instance
2. Instance methods
3. Scopes and class methods

In this case we should have written:

```
describe "validations" do
  context "adding self as a friend" do
    it "is invalid" do
      # write your expectation here -> this already tests for your friend_is_self private method
    end
  end
  
   describe "associations" do
      # you may use shoulda_matcher for this
    end
end
```